pkg/agent/plugin/nodeattestor/httpchallenge/httpchallenge_test.go

```diff
@@ -0,0 +1,209 @@
+//go:build !darwin
+
+package httpchallenge_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"testing"
+
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor"
+	nodeattestortest "github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/test"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/httpchallenge"
+	common_httpchallenge "github.com/spiffe/spire/pkg/common/plugin/httpchallenge"
+	"github.com/spiffe/spire/test/plugintest"
+	"github.com/stretchr/testify/require"
+)
+var (
+	streamBuilder = nodeattestortest.ServerStream("http_challenge")
+)
+
+func TestConfigureCommon(t *testing.T) {
+	tests := []struct {
+		name               string
+		hclConf            string
+		expErr             string
+	}{
+		{
+			name:    "Configure fails if receives wrong HCL configuration",
+			hclConf: "not HCL conf",
+			expErr:  "rpc error: code = InvalidArgument desc = unable to decode configuration",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			plugin := httpchallenge.New()
+
+			resp, err := plugin.Configure(context.Background(), &configv1.ConfigureRequest{HclConfiguration: tt.hclConf})
+			if tt.expErr != "" {
+				require.Contains(t, err.Error(), tt.expErr)
+				require.Nil(t, resp)
+				return
+			}
+
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+		})
+	}
+}
+
+func TestAidAttestationFailures(t *testing.T) {
+	tests := []struct {
+		name                              string
+		config                            string
+		expErr                            string
+		serverStream                      nodeattestor.ServerStream
+	}{
+		{
+			name:         "AidAttestation fails if server does not sends a challenge",
+			config:       "",
+			expErr:       "the error",
+			serverStream: streamBuilder.FailAndBuild(errors.New("the error")),
+		},
+		{
+			name:         "AidAttestation fails if agent cannot unmarshall server challenge",
+			config:       "",
+			expErr:       "rpc error: code = Internal desc = nodeattestor(http_challenge): unable to unmarshal challenge: invalid character 'o' in literal null (expecting 'u')",
+			serverStream: streamBuilder.IgnoreThenChallenge([]byte("not-a-challenge")).Build(),
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			p := loadAndConfigurePlugin(t, tt.config)
+
+			err = p.Attest(context.Background(), tt.serverStream)
+			if tt.expErr != "" {
+				require.Contains(t, err.Error(), tt.expErr)
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestAidAttestationSucceeds(t *testing.T) {
+	port := 9998
+        l, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
```

Comment:
If possible, we should ideally use an ephemeral port for this.

Reply:
hmm... yeah. good call.